terraform/README.md

```diff
@@ -0,0 +1,84 @@
+# PDS Validate ECS Task Definition Deployment
+
+The Terraform scripts in this directory deploy an ECS Task Definition for PDS Validate tool and a CloudWatch 
+log group for the ECS Task.
+
+
+## Prerequisites to Deploy Nucleus Baseline System
+
+1. An ECS Cluster should be available
+
+2. The docker image of the PDS Validate tool should be available in ECR 
+ - If an ECR image is not available, 
+   - Build a docker image using the instructions available in the docker directory for this repository
+   - Follow the instructions available as "View Push Commands" at https://us-west-2.console.aws.amazon.com/ecr/repositories/private/991026519024/pds-validate?region=us-west-2 
+
+3. Ability to get AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY and AWS_SESSION_TOKEN for the AWS account
+
+4. Terraform is installed in local environment (This was tested with Terraform v1.3.7. Any higher version should also work)
+- Instructions to install Terraform is available at https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli
+
+5. A VPC and one or more subnets should be available on AWS (obtain the VPC ID and subnet IDs from AWS console or from the AWS
+   system admin team of your AWS account)
+
+6. An EFS file system with an access point to read and write date
+
+
+## Steps to Deploy the PDS Nucleus Baseline System
+
+1. Clone the https://github.com/NASA-PDS/validate.git repository.
+
+```shell
+git clone https://github.com/NASA-PDS/validate.git
+```
+
+2. Open a terminal and change current working directory to the `nucleus/terraform` directory.
+
+```shell
+cd validate/terraform
+```
+
+3. Set following environment variables in terminal window
+    - AWS_ACCESS_KEY_ID
+    - AWS_SECRET_ACCESS_KEY
+    - AWS_SESSION_TOKEN
+    - AWS_DEFAULT_REGION
+
+4. Open the `terraform.tfvars` file locally and enter the value for following variables. Ensure these values match with your AWS Setup. Most of the below values can be obtained by the system admin team of your AWS account.
```

Comment:
@ramesh-maddegoda this works for now, but I think we should try to figure out a more robust solution using AWS Secrets per the same issue @sjoshi-jpl is encountering with his CI/CD work. We can revise this once we come up with a better solution.